cypress-io/cypress
# feat(app): adding navigation, pages, router, and layout

### Context
We need to support multiple pages and layouts in the app. This PR enables us to have routing, pages, and layouts. It adds a basic sidebar navigation component.

- Closes [UNIFY-308](https://cypress-io.atlassian.net/browse/UNIFY-308?atlOrigin=eyJpIjoiZTg2YTFlZDU2MDg4NDcxMWFjNjc1NzNiMWFiNTc2YzAiLCJwIjoiaiJ9)


### Naive Navbar UI
![Screen Shot 2021-09-22 at 2 41 11 AM](https://user-images.githubusercontent.com/2801156/134295975-65500d47-86a7-490d-bbc6-6469440aa218.png)

### Note
Currently, the schema stitching @tgriesser added is preventing the component tests from compiling.

### Features and bonuses
1. We have a router, with url-based navigation + back/forward support
2. We have layouts and pages.
3. Pages can have metadata (like a human-readable name) declared in `<route>` blocks
4. We have a 404 page :-)

### New UI
1. SidebarNavigation
2. SidebarNavigationRow

### How to test
1. `yarn dev --project packages/launchpad` from the workspace root
2. Launch a browser
3. Navigate to the /__vite__/ app path
4. See the router, layouts, etc and click through them to see the pages change
